### PR TITLE
Oregon Arcgis to STA

### PR DESCRIPTION
**Pull request by C-Loftus**

### CODE REVIEW

**Review by webb-ben — APPROVED** [review]

**Comment by C-Loftus** [comment]
@webb-ben just finished this. Tried to make very few changes to the other parts of the codebase and isolated changes into the `oregon/` folder. I think that is a good pattern. I tried to make the cli very simple. You just need to run `wis2box oregon load` in the container

File: wis2box/oregon/main.py [hunk 1]
```diff
@@ -0,0 +1,298 @@
...
+
+        all_stations = first_station_set["features"] + second_station_set["features"]
+        return all_stations
+
+    def _get_data_associated_with_station(
+        self, station_nbr, start_date, end_date, dataset
+    ) -> Tuple[list[float], list[str]]:
+        """Ingest all data from a station and return the third column."""
+        start_date = "09/25/2024 12:00:00"
+        end_date = "09/30/2024 12:00:00"
```
Comment (C-Loftus) [inline wis2box/oregon/main.py]: Currently hard coded for right now since I don't want us accidentally loading in all the data and spamming them

**Comment by C-Loftus** [comment]
May also make sense to try and merge any sta work into a central branch at some point so it is easier to add changes without needing to repeat them across branches.  (i.e. stuff like the docker volume for reloading or if i get pytest set up)

File: wis2box/oregon/main.py [hunk 2]
```diff
@@ -0,0 +1,298 @@
...
+
+THINGS_COLLECTION = "Things"
+
+
+class OregonStaRequestBuilder():
+
+    relevant_stations: List[int]
+
+    def __init__(self, relevant_stations: List[int]) -> None:
+        self.relevant_stations = relevant_stations
```
Comment (C-Loftus) [inline wis2box/oregon/main.py]: Let me know if you like the builder pattern or think there is a better way to structure this code. I also found this in RISE. Multi step ETL with nested json quickly gets kinda messy. That being said there isn't a great way to enforce order in instance methods so maybe there is a better way

**Comment by C-Loftus** [comment]
Added a helper code for storing the last download and updating to only add new data. I am just doing this by reading a file stored on disk. I think this is reasonable for the time being. This week it will be useful to align on our future plans for this branch and how it will be in relation to other branches.

I think I will probably need to write some pytest code for this but wanted to be aligned first in case any of this needs to be refactored

**Comment by C-Loftus** [comment]
Had some weird issues with aiohttp where the async functions were seeming to hang, but just switching to httpx and changing a few things syntax-wise seemed to fix it. Not really entirely clear what was going wrong. Load time down to 2m 15 seconds for one arbitrary station. 

Added a debugger which can be ran inside the container by doing stuff like this ` debugpy --wait-for-client  --listen 0.0.0.0:5678 $(which wis2box) oregon test -k "fully"`

Going to make the update functionality a bit more robust on monday. So there is a cache on disk where each station is associated with a time and that time is gotten by the tsv instead of what is passed into the CLI. Then once that is tested and stable I will get the crontab working and get it on google cloud. 

Once I get that done I think I will look into pagination on the hub2box UI. 

I also think there are some things that need to be cleaned up in wis2box, namely using Protocols and structural typing instead of using inheritance. In some cases the child class inherits from the parent in a non-typesafe way which makes it harder to know what is going on. I generally think that inheritance should be avoided if we can't use type checking with it. 

Just some thoughts after looking at this more.

**Pull request merged**